dissect/target/plugins/browsers/chromium.py

```diff
@@ -147,61 +154,162 @@ def downloads(self, browser_name: str = None) -> Iterator[BrowserDownloadRecord]
+                    for extension_id in extensions.keys():
+                        manifest = extensions.get(extension_id).get("manifest")
+                        if manifest:
+                            ts_install = webkittimestamp(extensions.get(extension_id).get("first_install_time"))
+                            ts_update = webkittimestamp(extensions.get(extension_id).get("last_update_time"))
+                            name = manifest.get("name")
+                            short_name = manifest.get("short_name")
+                            description = manifest.get("description")
+                            ext_path = extensions.get(extension_id).get("path")
+                            if ext_path and self.target.os == "windows":
+                                ext_path = path.from_windows(ext_path)
+                            elif ext_path:
+                                ext_path = path.from_posix(ext_path)
+                            ext_version = manifest.get("version")
+                            ext_permissions = manifest.get("permissions")
+                            manifest_version = manifest.get("manifest_version")
 
-    def _build_userdirs(self, hist_paths: list[str]) -> list[tuple]:
-        """Join the selected browser dirs with the user home path.
+                            if manifest.get("browser_action"):
+                                default_title = manifest.get("browser_action").get("default_title")
+                            else:
+                                default_title = None
+                        else:
+                            ts_install = None
+                            ts_update = None
+                            name = None
+                            short_name = None
+                            default_title = None
+                            description = None
+                            ext_version = None
+                            ext_path = None
+                            ext_permissions = None
+                            manifest_version = None
 
-        Args:
-            hist_paths: A list with browser paths as strings.
+                        yield self.BrowserExtensionRecord(
+                            ts_install=ts_install,
+                            ts_update=ts_update,
+                            browser=browser_name,
+                            id=extension_id,
+                            name=name,
+                            short_name=short_name,
+                            default_title=default_title,
+                            description=description,
+                            version=ext_version,
+                            ext_path=ext_path,
+                            from_webstore=extensions.get(extension_id).get("from_webstore"),
+                            permissions=ext_permissions,
+                            manifest_version=manifest_version,
+                            source=json_file.as_posix(),
```

Comment:
On second thoughts, I think for now we should go the long way here, so treat this similar to `ext_path`, e.g.:

```suggestion
                            if json_file and self.target.os == "windows":
                                json_file = path.from_windows(json_file)
                            elif json_file:
                                json_file = path.from_posix(json_file)

                            ...

                            source=json_file
```

This should actually be fixed in flow.record so we don't need to do this dance for every TargetPath type path (for stringy typed path you won't be able to circumvent explicitly casting unfortunately). I'm working on some updates in flow.record, so once they are in place we can remove the workarounds here.

As an additional note, I forgot that in the current context `source=json_file` would have resulted in similar behavior as `source=json_file.as_posix(). Just FYI.